core/domain/image_services_test.py

```diff
@@ -0,0 +1,128 @@
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for methods in the image_services."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import io
+import os
+
+from PIL import Image
+from PIL import ImageChops
+from core.domain import image_services
+from core.tests import test_utils
+import feconf
+import python_utils
+
+
+class ImageServicesUnitTests(test_utils.GenericTestBase):
+    """Tests for image_services."""
+
+    TEST_IMAGE_WIDTH = 3000
+    TEST_IMAGE_HEIGHT = 2092
+
+    def setUp(self):
+        super(ImageServicesUnitTests, self).setUp()
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'dummyLargeImage.jpg'),
+            mode='rb', encoding=None) as f:
+            self.jpeg_raw_image = f.read()
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'),
+            mode='rb', encoding=None) as f:
+            self.png_raw_image = f.read()
+
+    def test_image_dimensions_are_output_correctly(self):
+        height, width = (
+            image_services.get_image_dimensions(self.jpeg_raw_image))
+        self.assertEqual(self.TEST_IMAGE_HEIGHT, height)
+        self.assertEqual(self.TEST_IMAGE_WIDTH, width)
+
+    def test_compress_image_returns_correct_dimensions(self):
+        compressed_image = (
+            image_services.compress_image(self.jpeg_raw_image, 0.5))
+        height, width = (
+            image_services.get_image_dimensions(compressed_image))
+        self.assertEqual(self.TEST_IMAGE_HEIGHT * 0.5, height)
+        self.assertEqual(self.TEST_IMAGE_WIDTH * 0.5, width)
+
+    def test_invalid_scaling_factor_triggers_value_error(self):
+        value_exception = self.assertRaisesRegexp(
+            ValueError, r'Scaling factor should be in the interval \(0, 1].')
+        with value_exception:
+            image_services.compress_image(self.jpeg_raw_image, 1.1)
+
+    def test_compression_results_in_correct_format(self):
+        compressed_image = (
+            image_services.compress_image(self.jpeg_raw_image, 0.7))
+        pil_image = Image.open(io.BytesIO(compressed_image))
+        self.assertEqual(pil_image.format, 'JPEG')
+
+        compressed_image = (
+            image_services.compress_image(self.png_raw_image, 0.7))
+        pil_image = Image.open(io.BytesIO(compressed_image))
+        self.assertEqual(pil_image.format, 'PNG')
+
+    def test_compression_results_in_identical_files(self):
+        def is_identical(file1, file2):
+            """Checks the differences between the two images and returns if they
+            are different.
```

Comment:
Description does not match function name.

Docstring below too. Please fix.

Reply:
Done, removed the function per another comment.